### PR TITLE
Create a dedicated survey and tracking events for the CYS flow on core

### LINKED ISSUE #43667: [CYS on core] Create a dedicated survey and tracking events for the CYS flow on core.
On https://github.com/woocommerce/woocommerce/issues/41637 we temporarily removed the Survey button whenever the AI service is unavailable. This was a temporary change as we still didn't have the new survey for such cases ready by the time the CYS feature was released on WooExpress, but now we do.

With that said, the goal is to conditionally change the survey content depending on the environment the store owner is creating their site on (whether on WooExpress or Core) and also depending on the availability of AI on WooExpress: if AI is down, the survey should be the same as the one displayed on core.

## Designs

Figma: FlPl7fHaGv9mpu8Nfl0joH-fi-625_44310

### Updated Initial Transitional Page

* Ensure the survey button is displayed on core and also whenever AI is down on WooExpress.
* Update the copy for the body text (copy for the title and buttons remain the same):

| Initial Transitional Page > Expected Implementation on Core | Initial Transitional Page > Current view on WooExpress (should not be changed) |
|--------|--------|
|image | Screenshot 2024-01-15 at 13 34 38 |

### Updated Survey Form

| Survey Form > Expected Implementation on Core | Survey Form > Current view on WooExpress (should not be changed) |
|--------|--------|
| Screenshot 2024-01-15 at 11 41 34 | Screenshot 2024-01-15 at 11 38 18 |   

* Conditionally change the form content based on the environment, keeping the survey as-is on WooExpress and updated on Core.
* Open the designs on Figma to verify how the survey modal should look whenever filled.

### Updated tracking events

Currently, there are two events associated with the survey form:

* [`'customize_your_store_transitional_survey_click'`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-admin/client/customize-store/transitional/index.tsx#L106)
* [`'customize_your_store_transitional_survey_complete'`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-admin/client/customize-store/transitional/survey-form/index.tsx#L79)

For the implementation on core, we need to create new dedicated events to be triggered whenever the survey button is clicked, and feedback is given. Suggestion:

* `'customize_your_store_on_core_transitional_survey_click'`
* `'customize_your_store_on_core_transitional_survey_complete'`

Next, we need to also conditionally change the `recordEvent` based on the environment, keeping the events as-is on WooExpress and updated on Core.

### Updated transitional Page after survey complete

* Update the copy for the body text (copy for the title and buttons remain the same).

| Transitional Page after survey complete > Expected Implementation on Core | Transitional Page after survey complete > Current view on WooExpress (should not be changed) |
|--------|--------|
| image | Screenshot 2024-01-15 at 14 52 12|

### PR DESCRIPTION
**Pull request by albarin**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR:
- Changes the logic to show the survey button: it should always show unless the survey has already been completed.
- Updates the transitional page and survey copy when on Core or the AI is not available on WooExpress.

Closes https://github.com/woocommerce/woocommerce/issues/43667

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

**Local/JN site**
1. Head over to `/wp-admin/tools.php?page=woocommerce-admin-test-helper` and enable `customize-store` feature flag.
2. Go to `WooCommerce > Home`, click on the `Start Customizing` button, finish the CYS process, and click on `Done` to go to the `transitional` page.
3. Make sure the copy looks like this 👇 and you see the `Share feedback` button.
core
4. Click on the `Share feedback` button and check the survey looks like this:
surveycore

5. Complete the survey with some test data and send it, make sure the `Share feedback` button disappears.

------------------------------------------------------------------------

**WooExpress site**
1. Head over to `/wp-admin/tools.php?page=woocommerce-admin-test-helper` and enable `customize-store` feature flag.
2. Go to `WooCommerce > Home`, click on the `Start Customizing` button, finish the CYS process, and click on `Done` to go to the `transitional` page.
3. Make sure the copy looks like this 👇 and you see the `Share feedback` button.
express

4. Click on the `Share feedback` button and check the survey looks like this:
Screenshot 2024-01-25 at 14 49 43

5. Go back to `admin.php?page=wc-admin&path=%2Fcustomize-store`.
6. Open your browser's inspector and use Chrome's content override feature to change the content of `https://status.openai.com/api/v2/status.json` call. Change `status.indicator` to `major`.
7. Refresh the page, go through the process again, and click on `Done`.
8. Click on the `Share feedback` button and check the survey looks like the Core one:
surveycore

9. Complete the survey with some test data and send it, make sure the `Share feedback` button disappears.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Create a dedicated survey and tracking events for the CYS flow on core.

#### Comment 

</details>

### CODE REVIEW

**Review by gigitux — DISMISSED** [review]
Thanks for working on this! The copy looks okay! I noticed few things:

### Required Inputs

Looking at the Figma mockup that you shared, it looks like that:
- `On a scale of 1 = difficult to 5 = very easy, how would you rate the overall experience?`
- `What motivated you to choose the "Design your own theme" option?`
-  `Did you find anything confusing, irrelevant, or not useful?` 

questions should be required inputs. 

Also, it seems that the `*` should be red.

Currently, in production, only the stars are required in the field, so maybe it is not necessary to make any changes here.

### Tracking events

Regarding the events, do you think we should create dedicated events? I guess we should create a dedicated event at the beginning of the flow and keep everything as it is. In this way, we can easily distinguish the different sessions. I'm not an expert in this area, so maybe I'm wrong.

**Review by nefeline — COMMENTED** [review]
Thanks for your excellent work here, @albarin !

Test results:

> Make sure the copy looks like this 👇 and you see the Share feedback button.

✅ Confirmed the copy is correct and the content in the transitional page is accurate and matches the design:

Screenshot 2024-01-25 at 16 18 43

> Click on the Share feedback button and check the survey looks like this:

✅ Confirmed the copy is accurate and matches the design:

Screenshot 2024-01-25 at 16 18 59

> Complete the survey with some test data and send it, make sure the Share feedback button disappears.

✅ Confirmed that after the survey is complete, the share feedback button disappears:

Screenshot 2024-01-25 at 16 23 52

### Tracking events

✅ I've also confirmed that the scoped new tracking events are in place, including:

`'customize_your_store_on_core_transitional_survey_click'`
`'customize_your_store_on_core_transitional_survey_complete'`

Finally, I've also confirmed the `recordEvent` is conditionally changed based on the environment, keeping the events as-is on WooExpress and updated on Core.

### Notes

I noticed that we are allowing multiple check boxes to be selected in this section:

Screenshot 2024-01-25 at 16 22 45

Is this the expected? cc: @albarin @verofasulo 

Similar to what @gigitux mentioned, I also noticed that not all fields marked as required on the designs are required on the implementation:

Screenshot 2024-01-25 at 17 13 39

I believe we should change this to match the designs, WDYT?

**Comment by albarin** [comment]
> I noticed that we are allowing multiple check boxes to be selected in this section:

I think that's expected because they are checkboxes, not radio buttons, but let's see what @verofasulo says.

> Similar to what @gigitux mentioned, I also noticed that not all fields marked as required on the designs are required on the implementation:

I fixed this yesterday after @gigitux comment but never got to push the commit, I was convinced I did it, sorry about that 😅🤦‍♀️ Could you have another look @nefeline ?

**Comment by verofasulo** [comment]
> I think that's expected because they are checkboxes, not radio buttons, but let's see what @verofasulo says.

@albarin yes, that's correct 😄

File: plugins/woocommerce-admin/client/customize-store/assembler-hub/layout.tsx [hunk 1]
```diff
@@ -52,6 +52,7 @@ import { AiOfflineModal } from '~/customize-store/assembler-hub/onboarding-tour/
 import { useQuery } from '@woocommerce/navigation';
 import { FlowType } from '../types';
 import { isOfflineAIFlow } from '../guards';
+import { isWooExpress } from '~/utils/is-woo-express';
```
Comment (nefeline) [inline plugins/woocommerce-admin/client/customize-store/assembler-hub/layout.tsx:55]: For now, we can keep this conditional as-is. Still, it is worth mentioning that as a next step for this project, we are abstracting all references to WooExpress and AI from the core of WooCommerce (to be worked on as part of separate/upcoming PRs).

**Review by nefeline — APPROVED** [review]
> I think that's expected because they are checkboxes, not radio buttons

Makes sense! Thanks for clarifying @albarin and thanks for the confirmation, @verofasulo .

> I fixed this yesterday after @gigitux comment but never got to push the commit, I was convinced I did it, sorry about that 😅🤦‍♀️ Could you have another look @nefeline ?

Np at all! Thanks for taking care of this change @albarin : I've left a note regarding the conditionals (to be addressed as a next step for this project) & confirmed things are working as expected 🚀

**Pull request merged**